src/utils/pushforward.jl

```diff
@@ -1,53 +1,53 @@
 """
-    Pushforward{L,G}
+    Pushforward <: AbstractLayer
```

Comment:
Shouldn't `PushForward` be a `LossLayer`?
We always use it as a loss. We do not need to use it as an optimisation layer with perturbed because the `post_processing` can be included in the `maximizer` attribute of perturbed (I don't know about regularized though).

Reply:
It can be anything so that's why I hesitated. Ideally we would have a 4-way classification, between perturbed and regularized, and between probabilistic and deterministic, but that's not possible without traits